cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -2509,6 +2503,18 @@ private void processAsyncUpload(UploadToken uploadToken, UploadStatusCallback ca
         }
     }
 
+    private boolean shouldFinishExecution(boolean unsubscribe, Upload upload, UploadStatusCallback callback) {
```

Comment:
This method knows and does too much. Boolean methods should not have side effects, since they're basically a question. When I call a boolean method, I wouldn't expect something to happen. I would expect to just get an answer.
Also, `unsubscribe` is here to avoid having another `if` in the `processAsyncUpload` method, but the cost of this is that the method knows about the unsubscription mechanism and has a signature with too many parameters.